### ISSUE TITLE
Hot Reload fails with "ENC1002: The given key <assembly> was not present in the dictionary."

### ISSUE THREAD

**Issue by isadorasophia** [dotnet/roslyn#67497]
**Version Used**: 
Customers have reported this on Visual Studio 17.5 release.

**Steps to Reproduce**:
I am not able to reproduce this issue myself. However, I have seen a couple of different customer reports:

https://stackoverflow.com/questions/75020882/using-vs2022-i-cannot-change-c-sharp-code-in-debug-mode-enc1002
https://www.reddit.com/r/VisualStudio/comments/123wx63/edit_and_continue_stopped_working_in_most_or_all/

After attempting any code modification, the following error message:

`ENC1002 Cannot apply changes -- unexpected error: 'The given key <qualified-assembly-name> was not present in the dictionary.'`

**Diagnostic Id**: ENC1002

**Comment by isadorasophia** [comment]
+@tmat @davidwengier just in case anything similar to this issue comes to mind.

**Comment by davidwengier** [comment]
Thanks. I've not seen it before, but I asked for more info on Reddit, and for them to report a problem in VS.

**Comment by hendrikdevloed** [comment]
Same here. A simple edit in a .cs function, while on a breakpoint
```csharp
public partial class MainWindowViewModel {
private const int defaultPressure_5mmHg = 535;
// ... breakpoint somewhere in a private class method:
           int pressure = defaultPressure_5mmHg;
```
Changing it into `int pressure = defaultPressure_5mmHg-10;` and pressing Continue fails with a dialog "Edits were made which cannot be compiled".
Hot reload log:
>17:03 20.61 Invalid changes were found. Please refer to the Error List window to fix those issues.

Error list:
>Severity	Code	Description	Project	File	Line	Suppression State
>Error	ENC1002	Cannot apply changes -- unexpected error: 'The given key '_(my project name)_, Version=1.0.65535.65535, Culture=neutral, PublicKeyToken=null' was not present in the dictionary.'		_(my project name)_.sln	1	Active

**Comment by SylwesterZarebski** [comment]
I have the same error when making some very minor changes. Version of *1.0.65535.65535* looks like **AssemblyVersionAttribute** recompile bug, when version is set like *1.0.\**.

**Comment by davidwengier** [comment]
Using an '*' in an assembly version attribute will definitely prevent Edit and Continue from working. You should be able to wrap the attribute with an `#if !DEBUG` preprocessor block though.

**Comment by SylwesterZarebski** [comment]
Is there any work going on this issue?
In latest VS versions, it prevents using edit&continue in almost 100% libraries with `*.*` version.

**Comment by tmat** [comment]
Edit and Continue is not supported when using `*` in assembly version. A workaround is available:
https://github.com/dotnet/roslyn/blob/main/docs/wiki/EnC-Supported-Edits.md#projects-with-variable-assembly-versions

**Comment by SylwesterZarebski** [comment]
But that worked in previous versions of VS. It looks like a regression.

Workaround is not an option because changed numbering is **required** in DEBUG mode to detect new version by library loading and caching program. Numbering should be not touched when using E&C - as that was in previous versions of VS.
